mmcv/image/photometric.py

```diff
@@ -91,3 +91,51 @@ def posterize(img, bits):
     shift = 8 - bits
     img = np.left_shift(np.right_shift(img, shift), shift)
     return img
+
+
+def equalize(img):
+    """Equalize the image histogram. This function applies a non-linear mapping
+    to the input image, in order to create a uniform distribution of grayscale
+    values in the output image.
+
+    Args:
+        img (ndarray): Image to be equalized.
+
+    Returns:
+        ndarray: The equalized image.
+    """
+
+    def _scale_channel(im, c):
+        """Scale the data in the corresponding channel."""
+        im = im[:, :, c]
+        # Compute the histogram of the image channel.
+        histo = np.histogram(im, 256, (0, 255))[0]
+        # For computing the step, filter out the nonzeros.
+        nonzero_histo = histo[histo > 0]
+        step = (np.sum(nonzero_histo) - nonzero_histo[-1]) // 255
+
+        def _build_lut(histo, step, eps=1e-5):
+            # Compute the cumulative sum, shifted by step // 2
+            # and then normalized by step.
+            if step:
+                lut = (np.cumsum(histo) + (step // 2)) // step
+            else:
+                lut = (np.cumsum(histo) + (step // 2)) // (step + eps)
+            # Shift lut, prepending with 0.
+            lut = np.concatenate([[0], lut[:-1]], 0)
+            # Clip the counts to be in range.
+            return np.clip(lut, 0, 255)
+
+        # If step is zero, return the original image. Otherwise,
+        # build lut from the full histogram and step, and then
+        # index from it.
+        result = np.where(np.equal(step, 0), im, _build_lut(histo, step)[im])
```

Comment:
`_build_lut` is relatively short, and used only once. 
A function may not be necessary.